src/main/java/gmailgdogra/Officer.java

```diff
@@ -0,0 +1,73 @@
+package gmailgdogra;
+
+public class Officer {
+
+    private String firstName;
+    private String lastName;
+    private Location location;
+
+    public Officer(String firstName, String lastName, Location location) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.location = location;
+    }
+
+    public Officer(String firstName, String lastName) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public void setLastName(String lastName) {
+        this.lastName = lastName;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+
+    public void setLocation(Location location) {
+        this.location = location;
+    }
+
+    public String getFullName() {
+        return getFirstName() + " " + getLastName();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Officer officer = (Officer) o;
+
+        if (!getFirstName().equals(officer.getFirstName())) return false;
+        if (!getLastName().equals(officer.getLastName())) return false;
+        return getLocation() == officer.getLocation();
+    }
+
+    @Override
+    public int hashCode() {
+        int result = getFirstName().hashCode();
+        result = 31 * result + getLastName().hashCode();
+        return result;
```

Comment:
You use location to check equality but omit it when calculate hashcode. Technically, you don't break the equals/hashcode contract, but I hope you know what you are doing.

Reply:
I have updated hashCode and equals. both should be based on firstName and lastName only